internal/ansible/ansible.go

```diff
@@ -20,131 +20,260 @@ import (
-		WithStdoutCallback("json"),
-		// e.g options.AnsibleConnectionOptions{Connection: "local"}
-		WithConnectionOptions(&options.AnsibleConnectionOptions{}),
-		// e.g playbook.AnsiblePlaybookOptions{Inventory: "127.0.0.1,"}
-		WithOptions(&playbook.AnsiblePlaybookOptions{})), nil
+
+	return func(verbosity int) *exec.Cmd {
+		// check the verbosity since the exec.Command will fail if an arg as "" or " " be informed
+
+		cmdOptions := []string{
+			"--role", roleName,
+			"--roles-path", rolePath,
+			"--hosts", "localhost",
+		}
+		cmdArgs := []string{"run", wd}
+
+		if verbosity > 0 {
+			cmdOptions = append(cmdOptions, runnerutil.AnsibleVerbosityString(verbosity))
+		}
+		ansibleGathering := os.Getenv("ANSIBLE_GATHERING")
+
+		// When running a role directly, ansible-runner does not respect the ANSIBLE_GATHERING
+		// environment variable, so we need to skip fact collection manually
+		if ansibleGathering == "explicit" {
+			cmdOptions = append(cmdOptions, "--role-skip-facts")
+		}
+
+		// gosec is disabled here because of G204. We should pay attention that user can't
+		// make command injection via command argument
+		return exec.Command(runnerBinary, append(cmdArgs, cmdOptions...)...) //nolint:gosec
+	}, nil
+}
+
+// galaxyInstall Install non-exists collections with ansible-galaxy cli
+func (p Parameters) galaxyInstall() error {
+	galaxyBinary, err := galaxyutil.GalaxyBinary()
+	if err != nil {
+		return err
+	}
+
+	requirementsFilePath := runnerutil.GetFullPath(p.WorkingDir, galaxyutil.RequirementsFile)
+
+	cmdArgs := []string{"collection", "install"}
+	cmdOptions := []string{
+		"--requirements-file", requirementsFilePath,
+	}
+
+	// ansible-galaxy is by default verbose
+	cmdOptions = append(cmdOptions, "--verbose")
+
+	// gosec is disabled here because of G204. We should pay attention that user can't
+	// make command injection via command argument
+	dc := exec.Command(galaxyBinary, append(cmdArgs, cmdOptions...)...) //nolint:gosec
+
+	out, err := dc.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to install galaxy collections: %v: %v", string(out), err)
+	}
+	return nil
 }
 
-// NewAnsiblePlaybook returns a pbCmd that will be used as ansible-playbook client
-func NewAnsiblePlaybook(o ...PlaybookOption) *PbCmd {
+// Init initializes a new runner from parameters
+func (p Parameters) Init(ctx context.Context, cr *v1alpha1.AnsibleRun) (*Runner, error) {
+	if err := p.galaxyInstall(); err != nil {
+		return nil, err
+	}
+
+	addRolePlaybookPaths(p, cr)
 
-	pb := &playbook.AnsiblePlaybookCmd{
-		Playbooks: []string{},
+	if cr.Spec.ForProvider.Source == v1alpha1.ConfigurationSourceRemote {
+		// For inline mode playbook is stored in the predefined playbookYml file
+		// override user input if exists
+		cr.Spec.ForProvider.Playbook = runnerutil.PlaybookYml
+		cr.Spec.ForProvider.Role = runnerutil.RoleYml
 	}
 
+	var err error
+	var cmdFunc cmdFuncType
+	var path string
+	switch {
+	case cr.Spec.ForProvider.Playbook != "":
```

Comment:
For inline mode, which case block will be used to handle the case?

Reply:
So after the last fix, in the inline mode we only deal with `Playbook`, so it will fallback to this case => `cr.Spec.ForProvider.Playbook`